lib/dayron/repo.ex

```diff
@@ -167,84 +167,208 @@ defmodule Dayron.Repo do
     end
   end
 
   @doc false
   def get!(adapter, model, id, opts, config) do
-    case get(adapter, model, id, opts, config) do
-      nil ->
-        url = Config.get_request_url(config, model, [id: id])
-        raise Dayron.NoResultsError, method: "GET", url: url
-      model -> model
+    {request, response} = config
+      |> Config.init_request_data(:get, model, id: id)
+      |> execute!(adapter, opts, config)
+
+    case response do
+      %HTTPoison.Response{status_code: 200, body: body} ->
+        Model.from_json(model, body)
+      %HTTPoison.Response{status_code: code} when code >= 300 and code < 500 ->
+        raise Dayron.NoResultsError, method: request.method, url: request.url
     end
   end
 
   @doc false
   def all(adapter, model, opts, config) do
-    {url, response} = get_response(adapter, model, [], opts, config)
+    {_request, response} = config
+      |> Config.init_request_data(:get, model)
+      |> execute!(adapter, opts, config)
+
     case response do
       %HTTPoison.Response{status_code: 200, body: body} ->
         Model.from_json_list(model, body)
-      %HTTPoison.Response{status_code: code, body: body} when code >= 300 ->
-        raise Dayron.ServerError, method: "GET", url: url, body: body
-      %HTTPoison.Error{reason: reason} ->
-        raise Dayron.ClientError, method: "GET", url: url, reason: reason
     end
   end
 
   @doc false
   def insert(adapter, model, data, opts, config) do
-    {url, response} = post_response(adapter, model, data, opts, config)
+    {request, response} = config
+      |> Config.init_request_data(:post, model, body: data)
+      |> execute!(adapter, opts, config)
+
     case response do
       %HTTPoison.Response{status_code: 201, body: body} ->
         {:ok, Model.from_json(model, body)}
       %HTTPoison.Response{status_code: 422, body: body} ->
-        {:error, %{method: "POST", url: url, response: body}}
-      %HTTPoison.Response{status_code: code, body: body} when code >= 500 ->
-        raise Dayron.ServerError, method: "POST", url: url, body: body
-      %HTTPoison.Error{reason: reason} ->
-        raise Dayron.ClientError, method: "POST", url: url, reason: reason
+        {:error, %{method: request.method, url: request.url, response: body}}
     end
   end
 
+  @doc false
   def insert!(adapter, model, data, opts, config) do
     case insert(adapter, model, data, opts, config) do
       {:ok, model} -> {:ok, model}
       {:error, error} -> raise Dayron.ValidationError, Map.to_list(error)
     end
   end
 
-  defp get_response(adapter, model, url_opts, request_opts, config) do
-    url = Config.get_request_url(config, model, url_opts)
-    headers = Config.get_headers(config)
-    {_, response} = adapter.get(url, headers, request_opts)
-    if Config.log_responses?(config) do
-      ResponseLogger.log("GET", url, response)
+  @doc false
+  def update(adapter, model, data, opts, config) do
+    {request, response} = config
```

Comment:
minor: `When assigning to a multi-line call, begin a new line after the =` from https://github.com/rrrene/elixir-style-guide